ToolChain/Assembler/Parser/Parser.py

```diff
@@ -20,544 +19,368 @@
-from CapuaEnvironment.Instruction.OperationDescription import operationDescription
-from CapuaEnvironment.IntructionFetchUnit.FormDescription import formDescription
-from Configuration.Configuration import REGISTER_A,  \
-                                        REGISTER_B,  \
-                                        REGISTER_C,  \
-                                        REGISTER_D,  \
-                                        REGISTER_E,  \
-                                        REGISTER_F,  \
-                                        REGISTER_G,  \
+__author__ = "CSE"
+__copyright__ = "Copyright 2018, CSE"
+__credits__ = ["CSE"]
+__license__ = "GPL"
+__version__ = "2.1"
+__maintainer__ = "CSE"
+__status__ = "Dev"
+
+from Configuration.Configuration import REGISTER_A, \
+                                        REGISTER_B, \
+                                        REGISTER_C, \
+                                        REGISTER_D, \
+                                        REGISTER_E, \
+                                        REGISTER_F, \
+                                        REGISTER_G, \
                                         REGISTER_A2, \
                                         REGISTER_B2, \
                                         REGISTER_C2, \
                                         REGISTER_D2, \
                                         REGISTER_E2, \
                                         REGISTER_F2, \
                                         REGISTER_G2, \
-                                        REGISTER_S,  \
-                                        REGISTER_S2
-from ToolChain.Assembler.Constants import REGISTER_PREFIX, \
-                                          IMMEDIATE_PREFIX, \
-                                          WIDTH_INDICATORS, \
-                                          FLAGS_INDICATORS, \
-                                          COMMENT_INDICATORS, \
-                                          MEMORY_REFERENCE_INDICATORS, \
-                                          DATA_ALPHA_INDICATOR, \
-                                          DATA_NUMERIC_INDICATOR, \
-                                          DATA_MEMORY_REFERENCE, \
-                                          EXPORTED_REFERENCE_INDICATOR
+                                        REGISTER_S, \
+                                        REGISTER_S2, \
+                                        INSTRUCTION_LIST, \
+                                        IDENTIFIER_LIST, \
+                                        LABEL_INSTRUCTIONS
 
-import struct
+from CapuaEnvironment.IntructionFetchUnit.FormDescription import formDescription
+from CapuaEnvironment.Instruction.OperationDescription import operationDescription
 
-__author__ = "CSE"
-__copyright__ = "Copyright 2018, CSE"
-__credits__ = ["CSE"]
-__license__ = "GPL"
-__version__ = "2.0"
-__maintainer__ = "CSE"
-__status__ = "Dev"
+import struct
```

Comment:
This might be misleading since things have been shifted around, but import struct is directly under the rest of the imports in the new file:

__author__ = "CSE"
__copyright__ = "Copyright 2018, CSE"
__credits__ = ["CSE"]
__license__ = "GPL"
__version__ = "2.1"
__maintainer__ = "CSE"
__status__ = "Dev"

from Configuration.Configuration import REGISTER_A, \
                                        REGISTER_B, \
                                        REGISTER_C, \
                                        REGISTER_D, \
                                        REGISTER_E, \
                                        REGISTER_F, \
                                        REGISTER_G, \
                                        REGISTER_A2, \
                                        REGISTER_B2, \
                                        REGISTER_C2, \
                                        REGISTER_D2, \
                                        REGISTER_E2, \
                                        REGISTER_F2, \
                                        REGISTER_G2, \
                                        REGISTER_S, \
                                        REGISTER_S2, \
                                        INSTRUCTION_LIST, \
                                        IDENTIFIER_LIST, \
                                        LABEL_INSTRUCTIONS

from CapuaEnvironment.IntructionFetchUnit.FormDescription import formDescription
from CapuaEnvironment.Instruction.OperationDescription import operationDescription

import struct


class Parser:

Reply:
You are right. 

The but the assembler is still missing the copyright etc [here](https://github.com/kommisar5150/spartacus/blob/906c993cda340c3e6107debe6ea2d4e6d59d132e/ToolChain/Assembler/Assembler.py)